modules/Link.js

```diff
@@ -62,6 +77,60 @@ const Link = React.createClass({
     }
   },
 
+  getInitialState() {
+    return {
+      isActive: isLinkActive(this.context.router, this.props)
+    }
+  },
+
+  componentWillMount() {
+    this._isActive = this.state.isActive
```

Comment:
We’re going to use `this._isActive` for the pending state from now on. This lets us quickly bail out if it hasn’t changed. (See `updateIsActive` later.) This is a performance optimization.